### ISSUE TITLE
Batch Import not working

### ISSUE THREAD

**Issue by aosti** [NationalSecurityAgency/ghidra#1235]
**Describe the bug**
I have a folder full of binary files (.bin) that contain mips instructions and want to perform batch import on Ghidra. However, after selecting them, I'm shown the screen as contained in the screenshot section.

**To Reproduce**
1. Execute ./ghidraRun
2. Click on File > Batch Import
3. Select the folder containing the binary files
4. See unexpected behauvior.

**Expected behauvior**
The list "Files to Import" get filled with the 276 files in the folder.

**Screenshots**
![image](https://user-images.githubusercontent.com/8411730/68698504-c0a08a00-055f-11ea-840b-5d258a3188fa.png)

**Environment (please complete the following information):**
 - OS: macOS 10.15.1
 - Java Version: 12.0.1
 - Ghidra Version: 9.1

**Comment by dragonmacher** [comment]
Has this feature worked for you on this directory in a previous version?

Have you tried changing the 'Depth limit' value?

**Comment by aosti** [comment]
Changing the depth limit for bigger values has no effect. Note that single importing as raw binary works.  I'll do a test with older versions.

**Comment by dragonmacher** [comment]
I would expect this to work.  We will have to see if we can re-create this locally.

**Comment by aosti** [comment]
I'll attach some files to this ticket.

**Comment by dev747368** [comment]
This is working as designed.

The batch importer currently filters out anything that doesn't have a defined loader, which includes raw binary files.  It does this to limit the spam that random binary / text / etc files would cause.

I could see how this would useful option though, if you were importing a container that you were sure about.  I'll transform this issue into an enhancement request.

**Comment by soutzis** [comment]
Since this is still, open I'd like to add my 2 cents. I have a folder full of raw (malware) binaries, along with an associated report in .json format,
The filenames are the sha-256 digest of the binary and the reports just have the additional .json extension.

Example:
raw file -> 0ddee76c519101aa4ded7546408f51465bf2adaf9b584ad5fb87845583e47276
associated report -> 0ddee76c519101aa4ded7546408f51465bf2adaf9b584ad5fb87845583e47276.json

A very useful **enhancement**, would be to have the option to choose what filetypes to exclude, while importing everything else (considering that the user is sure of what they are batch-importing). In my case, I could exclude any '.json' file only

**Comment by dev747368** [comment]
@soutzis yes, that sounds like a previously reported bug (#1572).  One of the file-format loaders, when probing one of the binaries, is having a problem (that it should be catching on its own).  

In this case, your enhancement request isn't 100% clear.  Do you suspect that the error is caused by the .json file?  My guess is that it probably is not causing the error.  Its probably a malformed PE or ELF file.

**Comment by soutzis** [comment]
@dev747368 Most likely the error is not caused by the .json file. Nevertheless, being able to exclude file types during the batch import process could be still beneficial. For example, it would be clearer that the malformed binary is causing the error.

**Comment by dev747368** [comment]
Well, you can exclude files types.... after the initial scan and you've got your list, before you hit the start button, you can check / uncheck individual rows in the table to include / exclude that row (which is grouped by file type & loader & lang).

**Comment by mikeage** [comment]
> The batch importer currently filters out anything that doesn't have a defined loader, which includes raw binary files. It does this to limit the spam that random binary / text / etc files would cause.

I just came across this issue; I, too, would also like to be able to Batch Import many Raw Binaries.

I understand that this isn't supported currently, from the snippet I quoted, but from https://www.ghidradocs.com/12.1_PUBLIC/help/Base/help/topics/ImporterPlugin/importer.htm, it wasn't clear to me that only files with a known loader would be importable. Perhaps the docs should be updated accordingly?

**Comment by ryanmkurtz** [comment]
Our team discussed this and decided that adding the support to the batch importer is currently not worth the time investment, since a similar result can be achieved with the headless analyzer.
